api_confluence.py

```diff
@@ -209,8 +239,169 @@ def pages():
+    content for rewriting pages or creating new XML template configs.
+    """
+
+    TEMPLATE_PATH = f"{PATH_XML_FILES}/build-validation.xml"
+
+    # XML template
+    # with open(TEMPLATE_PATH, "r", encoding="utf-8") as file:
+    #     xml_content = file.read()
+    output_xml = render_xml_template(TEMPLATE_PATH, params)
+
+    # Confluence page (for updating)
+    page_url = url_page(page_id)
+
+    # Write output_xml to page
+    print(f"UPDATING PAGE - page_id: {page_id}")
+    page_data = page_object(page_url)
+    current_version = page_data["version"]["number"]
+
+    payload = page_payload(page_id, page_title, page_data, current_version, output_xml) # noqa
+    page_payload_write(page_id, payload)
+
+
+# REPORT: Build Validation
+
+def page_report_build_validation(
+    page_id,
+    projects_id,
+    testrail_milestone_id,
+    testrail_milestone_title,
+    testrail_report_url,
+    build_version,
+    signoff_date,
+    test_status,
+    test_summary,
+    ship_recommend,
+    ship_recommend_verbose,
+    contacts,
+):
+    """
+    Applies params to jinja/mustache-style XML template and inserts into page
+    """
+
+    # TODO: hard-coded params for now...
+    params = {
+        'release_date': signoff_date,
+        'build_version': build_version,
+        'build_version_commit_hash': '12345',
+        'testrail_report_url': 'https://mozilla.testrail.io/index.php?/runs/view/108599&group_by=cases:section_id&group_order=asc', # noqa
+        'testrail_report_summary': test_summary,
+        'testrail_run_1_result': 'PASS3D',
+        'testrail_run_1_build_type': 'ARM64',
+        'testrail_run_1_device': 'iPhone 16 Pro (iOS 18.3.2)',
+        'testrail_run_2_result': 'PASS3D',
+        'testrail_run_2_build_type': 'ARM64',
+        'testrail_run_2_device': 'iPhone 16 Pro (iOS 18.3.2)',
+        'testrail_issues_verified': 'N/A',
+        'testrail_issues_new': 'N/A',
+        'testrail_issues_known': '<a href="https://mozilla-hub.atlassian.net/browse/FXIOS-12650"</a>', # noqa
+        'testrail_contacts': contacts,
+    }
+
+    params = {
+        **locals()
+    }
+
+    TEMPLATE_PATH = f"{PATH_XML_FILES}/build-validation.xml"
+
+    page_url = url_page(page_id)
+    # page_url_storage = url_page_content_storage(page_id)
+    with open(TEMPLATE_PATH, "r", encoding="utf-8") as file:
+        xml_content = file.read()
+    output_xml = render_xml_template(TEMPLATE_PATH, params)
+
+    print(f"UPDATE PAGE - page_id: {page_id}")
+    page_data = page_object(page_url)
+    current_version = page_data["version"]["number"]
+
+    # new_content = page_html(page_id, page_sections)
+    payload = page_payload(page_id, page_title, page_data, current_version, output_xml) # noqa
+    page_payload_write(page_id, payload)
+
+
 def main():
-    pages()
+    # TODO: phase 2 PR - instead of invoking this directly from main,
+    # invoke it from __main__.py --report-type looker-graphs
+    pages_looker_graphs()
```

Comment:
When the daily confluence update [github action job](https://github.com/mozilla-mobile/testops-dashboard/blob/master/.github/workflows/confluence-daily.yaml#L45C14-L45C41) runs, this main() will get executed same as before.  The looker graph execution function has been renamed to:  pages_looker_graphs() and will get invoked here.  Everything below line 370 is phase 2 (hence, commented out)